google-cloud-firestore/lib/google/cloud/firestore/convert.rb

```diff
@@ -263,10 +262,8 @@ def write_for_set_merge doc_path, data, field_paths, allow_empty
             # Restore delete paths
             field_paths += delete_field_paths_and_values.keys
 
-            if data.empty? && !allow_empty
-              if field_paths_and_values.empty? && delete_field_paths_and_values.empty?
-                raise ArgumentError, "data required for set with merge"
-              end
+            if data.empty? && !allow_empty && (field_paths_and_values.empty? && delete_field_paths_and_values.empty?)
```

Comment:
I don't think the parens are necessary. Logical AND is associative.

Reply:
Done, thanks!